File: golang/0443.go
func compress(chars []byte) int {
	debug.SetGCPercent(1)
	res := 0
	i := 0
	length := len(chars)
	for i < length {
		groupLength := 1
		for i+groupLength < length && chars[i+groupLength] == chars[i] {
			groupLength++
		}
		chars[res] = chars[i]
		res++
		if groupLength > 1 {
			l := strconv.Itoa(groupLength)
			for _, c := range l {
				chars[res] = byte(c)
				res++
			}
		}
		i += groupLength
	}
	return res
}

File: rust/0443.rs
impl Solution {
    pub fn compress(chars: &mut Vec<char>) -> i32 {
        let mut index = 0;
        let mut res = 0;
        while index < chars.len() {
            let mut group_length = 1;
            while index + group_length < chars.len() && chars[index + group_length] == chars[index] {
                group_length += 1;
            }
            chars[res] = chars[index];
            res += 1;
            if group_length > 1 {
                for c in group_length.to_string().chars() {
                    chars[res] = c;
                    res += 1;
                }
            }
            index += group_length;
        }
        res as i32
    }
}
